### PR TITLE
add transaction batching

### PR DESCRIPTION
**Pull request by none00y**
# Changes
- added transaction batching to sdk

### CODE REVIEW

File: sdk/tests/batch.test.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+    invariant = await Invariant.deploy(api, admin, toPercentage(1n, 2n))
+    token0Address = await FungibleToken.deploy(api, admin, 'Coin', 'COIN', 0n)
+    token1Address = await FungibleToken.deploy(api, admin, 'Coin', 'COIN', 0n)
+  })
+
+  it('batch deposit and withdraw', async function () {
+    this.timeout(20000)
+    await batchTxs(api, admin, [
+      await GRC20.mintTx(admin.addressRaw, 1000000000000n, token0Address),
```
Comment (wojciech-cichocki) [inline sdk/tests/batch.test.ts:42]: Is the use of `await` inside `batchTxs` required?

File: sdk/tests/batch.test.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        await GRC20.balanceOf(invariant.programId(), token0Address),
+        1000000000000n
+      )
+      assert.strictEqual(
+        await GRC20.balanceOf(invariant.programId(), token1Address),
+        1000000000000n
+      )
+    }
+
+    await batchTxs(api, admin, [
```
Comment (wojciech-cichocki) [inline sdk/tests/batch.test.ts:68]: What information about the success of invoking this batch will be possible to obtain?

File: sdk/tests/batch.test.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        await GRC20.approveTx(invariant.programId(), 1000000000000n, token1Address),
+        await invariant.depositSingleTokenTx(token0Address, 1000000000000n),
+        await invariant.depositSingleTokenTx(token1Address, 1000000000000n)
+      ])
+    } catch (e) {
+      assert(e instanceof BatchError)
+      assert.deepEqual(
+        e.failedTxs,
+        new Map([
+          [0, 'Panic occurred: Underflow'],
```
Comment (wojciech-cichocki) [inline sdk/tests/batch.test.ts:102]: What information about the success of invoking this batch will be possible to obtain?

File: sdk/tests/batch.test.ts [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+          [1, 'Panic occurred: Underflow'],
```
Comment (wojciech-cichocki) [inline sdk/tests/batch.test.ts:103]: Additionally, does the batch here have a guarantee of sequential execution, and will the flow be interrupted if one of the transactions fails?

File: sdk/tests/batch.test.ts (hunk 1 above)
Comment (none00y) [reply sdk/tests/batch.test.ts:42]: Yes, setting the gas limit requires using await

**Review by wojciech-cichocki — APPROVED** [review]

**Pull request merged**